### PR TITLE
Claimset management design notes

### PR DESCRIPTION
**Pull request by stephenfuqua**

### CODE REVIEW

File: docs/CS/CLAIMSET-MGMT.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+  * /claimsets/{id}/resourceClaimActions
+    * POST
+    * PUT
+    * DELETE
+    * /claimsets/{id}/resourceClaimActions/{id}
+      * POST overrideAuthorizationStrategy
+      * POST resetAuthorizationStrategies
+* /resourceClaims
+  * GET all
+  * GET by id
```
Comment (stephenfuqua) [inline docs/CS/CLAIMSET-MGMT.md:63]: Are we really sure that the CLOB payload includes all of the child resource claims? For example, in the "managed descriptors" payload, does it really include a list of all of the descriptors?

File: docs/CS/CLAIMSET-MGMT.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+In the claimset GET requests, the readonly `_applications` array implies that
+the query needs to join to the `Application` table. Thus the `Application` table
+needs an index on `ClaimSetName`.
+
+* `GET /v2/claimSets/{id}/export`
+* `GET /v2/claimSets`
+  * Return the simple response body defined in the Admin API specification.
+  * Includes paging operations.
+* `GET /v2/claimSets?hydrated=true`
```
Comment (stephenfuqua) [inline docs/CS/CLAIMSET-MGMT.md:124]:
```suggestion
* `GET /v2/claimSets?verbose=true`
```

File: docs/CS/CLAIMSET-MGMT.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+  * Returns the entire CLOB, with the same response as the `/export` endpoint.
+  * Needs to support the normal paging operations, though `offset` and `limit`
+    should _not_ be required fields.
+* `GET /v2/claimSets/{id}`
+  * Return the simple response body defined in the Admin API specification.
+* `GET /v2/claimSets/{id}?hydrated=true`
```
Comment (stephenfuqua) [inline docs/CS/CLAIMSET-MGMT.md:130]:
```suggestion
* `GET /v2/claimSets/{id}?verbose=true`
```

File: docs/CS/CLAIMSET-MGMT.md [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+## REST API Implications
+
+The DMS Configuration Service will not, initially, support all of the Admin API
+endpoints. Further discussion with the affected community members is needed to
+determine if this is a viable long-term direction. Initially, the key goal is to
+enable the DMS to retrieve claimset information from the Configuration Service.
+This is essentially the same as the export operation. We will modify the GET all
+and GET by id endpoints to accept a query parameter that returns the "fully
+hydrated" claimsets.
```
Comment (stephenfuqua) [inline docs/CS/CLAIMSET-MGMT.md:105]:
```suggestion
hydrated" claimsets (with query string parameter `verbose`).
```

File: docs/CS/CLAIMSET-MGMT.md [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+* `DELETE /v2/claimSets/{id}`
+* `POST /v2/claimSets/copy`
+* `POST /v2/claimSets/import`
+  * Becomes a synonym to `/v2/claimSets`, except that now the `resourceClaims`
+    array will be a required attribute.
+
+#### AuthorizationStrategies
+
+Support full CRUD operations on `/v2/authorizationStrategies`, even though the Admin API
+specification only includes the read operations.
```
[the remark this answers is not in the document]
Comment (stephenfuqua) [reply docs/CS/CLAIMSET-MGMT.md:160]: Just hard-code the list of constants. There is no reason to allow updates in a table. Similar to `/v2/actions`. 1 point.

File: docs/CS/CLAIMSET-MGMT.md [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+
+## Summary of Changes Compared to Admin API 2.2 Specification
+
+* `offset` and `limit` should not be required parameters
+* `PUT` requests will require the presence of the matching `id` attribute in the
+  request body.
+* Both `POST` and `PUT` on `/v2/claimSets` will support use of the
+  `resourceClaims` attribute.
+* A `GET` request to `/v2/claimSets` or `/v2/claimSets/{id}` will return the
+  `resourceClaims` array _if_ query string parameter `hydrated` is received with
```
Comment (stephenfuqua) [inline docs/CS/CLAIMSET-MGMT.md:188]:
```suggestion
  `resourceClaims` array _if_ query string parameter `verbose` is received with
```

**Review by bradbanister — APPROVED** [review]

**Pull request merged**